controller/controller.go

```diff
@@ -67,29 +67,19 @@ func (c *controller) Start() error {
 		return fmt.Errorf("unable to start load balancer: %v", err)
 	}
 
-	c.watcher = k8s.NewWatcher()
-	err = c.client.WatchIngresses(c.watcher)
-	if err != nil {
-		return fmt.Errorf("unable to watch ingresses: %v", err)
-	}
-
+	c.watcher = c.client.WatchIngresses()
 	go c.watchForUpdates()
 
 	c.started = true
 	return nil
 }
 
 func (c *controller) watchForUpdates() {
-	for {
-		select {
-		case <-c.watcher.Done():
-			return
-		case <-c.watcher.Updates():
-			log.Info("Received update on watcher")
-			err := c.updateLoadBalancer()
-			if err != nil {
-				log.Errorf("Unable to update load balancer: %v", err)
-			}
+	for _ = range c.watcher.Updates() {
+		log.Info("Received update on watcher")
+		err := c.updateLoadBalancer()
```

Comment:
can we rename 'updateLoadBalancer'  to just 'update' or 'doUpdate', but it's not load balancer specific anymore